library/psa_crypto.c

```diff
@@ -2871,21 +2871,26 @@ static psa_status_t psa_sign_internal( mbedtls_svc_key_id_t key,
 
     *signature_length = 0;
 
-    if( operation == PSA_SIGN_INVALID )
-        return( PSA_ERROR_INVALID_ARGUMENT );
-    else
+    switch( operation )
     {
-        if( ! PSA_ALG_IS_SIGN_MESSAGE( alg ) )
-            return( PSA_ERROR_INVALID_ARGUMENT );
+        case PSA_VERIFY_HASH:
+            if( ! PSA_ALG_IS_HASH_AND_SIGN( alg ) )
+                return( PSA_ERROR_INVALID_ARGUMENT );
+            break;
```

Comment:
Does the hash component of the hash-and-sign algorithm not need to be a valid hash algorithm when asking to sign the hash? I believe there should be a check somewhere matching the input length to the expected hash length. What do you think, @gilles-peskine-arm?

Reply:
According to the PSA specification:

> Some hash-and-sign mechanisms apply a padding or encoding to the hash. In such cases, the encoded hash must be passed to this function. The current version of this specification defines one such signature algorithm: PSA_ALG_RSA_PKCS1V15_SIGN_RAW.

It is only needed when PSA_ALG_RSA_PKCS1V15_SIGN_RAW is used.